dissect/ffs/ffs.py

```diff
@@ -143,12 +147,13 @@ def inode_allocated(self, inum):
 
 
 class INode:
-    def __init__(self, fs, inum, name=None, filetype=None):
+    def __init__(self, fs, inum, name=None, filetype=None, parent=None):
         self.fs = fs
         self.inum = inum
         self.name = name
         self._type = filetype
-
+        self._link_inode = None
+        self.parent = parent
```

Comment:
```suggestion
        self.parent = parent

```
Nitpicky, but I'd like to keep setting attributes from constructor arguments under each other, and I believe `_link_inode` can actually become a `@cached_property`.